### PR TITLE
support binary protocol version 0.11

### PR DESCRIPTION
**Pull request by fmoor**
Increases the number of bytes dedicated to the object descriptor element
flag field from 8 to 32 bytes.

server change https://github.com/edgedb/edgedb/pull/2603

### CODE REVIEW

File: edgedb/protocol/protocol.pyx [hunk 1]
```diff
@@ -132,7 +132,7 @@ cdef class SansIOProtocol:
 
         self.server_settings = {}
         self.reset_status()
-        self.protocol_version = (PROTO_VER_MAJOR, 0)
+        self.protocol_version = (PROTO_VER_MAJOR, PROTO_VER_MINOR)
```
Comment (tailhook) [inline edgedb/protocol/protocol.pyx:135]: I think this is bad to assume highest protocol version before negotiation is done. And I don't see any need to change it in this PR.

Comment (fmoor) [reply edgedb/protocol/protocol.pyx]: changed

File: tests/test_datetime.py [hunk 2]
```diff
@@ -62,8 +62,11 @@ async def test_duration_01(self):
         ''', durs)
         self.assertEqual(list(durs_from_db), durs)
 
-    @unittest.expectedFailure
     async def test_relative_duration_01(self):
+        try:
+            self.con.query("SELECT <cal::relative_duration>'1y'")
+        except errors.InvalidReferenceError:
+            self.skipTest("feature not implemented")
```
Comment (fmoor) [inline tests/test_datetime.py:69]: Is there a better way to skip this test only on the stable version and not nightly? Or rather only for the server versions that don't support it?

**Comment by fmoor** [comment]
I forgot to separate the flags and cardinality enum. Fixing WIP.

File: tests/test_datetime.py (hunk 2 above)
Comment (1st1) [reply tests/test_datetime.py:69]: Well, you can introspect the DB and check if the `cal::relative_duration` type is defined. If not - skip the test. We generally trust the introspection. Alternatively, you can query the server version with https://www.edgedb.com/docs/edgeql/funcops/sys#function::sys::get_version

**Review by 1st1 — APPROVED** [review]

**Pull request merged**